src/libertem/udf/base.py

```diff
@@ -982,12 +1027,13 @@ def _init_udfs(self, numpy_udfs, cupy_udfs, partition, roi, corrections, device_
             udf.set_meta(meta)
         return (meta, tiling_scheme, dtype)
 
-    def _run_tile(self, udfs, partition, tile, device_tile):
+    def _run_tile(self, udfs, partition, tile, device_tile, roi):
         for udf in udfs:
             method = udf.get_method()
             if method == 'tile':
                 udf.set_contiguous_views_for_tile(partition, tile)
                 udf.set_slice(tile.tile_slice)
+                udf.set_coordinates(tile.tile_slice, partition.meta.shape, roi)
                 udf.process_tile(device_tile)
             elif method == 'frame':
```

Comment:
There needs to be some special coordinate handling for `method == 'frame'` here, I think, maybe generate the whole coordinate list for the `tile` once, and then index into that with `frame_idx` in the inner loop?

Reply:
Okay, the new coords generation code is out of the UDF interface and I've added a test for `method == 'frame'` too.